vocabularies/Hierarchy.xml

```diff
@@ -45,27 +45,133 @@
       </Term>
       <ComplexType Name="RecursiveHierarchyType">
         <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.LongDescription">
+          <String>The nullable property paths in this complex type refer to properties of an entity type that contain
+information about a node in a recursive hierarchy. When hierarchical transformations
+are applied whose first parameter has the annotated entity type
+and whose second parameter is the annotation qualifier,
+then these properties are derived for each node in the result set.
+
+For requests like
+```
+SalesOrganizations?$apply=
+descendants(..., filter(ID eq 'US'), keep start)
+/ancestors(..., filter(contains(Name, 'New York'))
+/Hierarchy.TopLevels(..., Levels=2)
+/top(10)
+```
+the following collections are distinguished:
+- sub-hierarchy: the output set of a `descendants` transformation filtering a single node value:
+  US sales organizations (first two rows of request)
+- [matching](#MatchCount) nodes: US sales organizations with "New York" in their name
+  (output set of `filter` transformation in third row of request)
+- unlimited hierarchy: the output set of the last `ancestors`, `descendants` or `traverse` transformation
+  disregarding numeric fifth parameters: (first three rows of request)
+  `descendants(..., filter(ID eq 'US'), keep start)/ancestors(..., filter(contains(Name, 'New York')))`
+- limited hierarchy: the output set of the last `ancestors`, `descendants`, `traverse` or
+  [`Hierarchy.TopLevels`](#TopLevels) transformation: (first four rows of request)</String>
```

Comment:
These definitions, on which the properties of `RecursiveHierarchy` are based, are somewhat fluffy and reflect the thinking of OData V2 hierarchies, where the following kinds of `$filter` were supported:
- filter on node ID: determines a sub-hierarchy
- filter on parent ID: determines children of a node
- filter on `hierarchy_level`: limits the hierarchy
- filter on other property: determines matching nodes.

These are also the kinds of filter that the underlying database supports. OData V4 goes conceptually beyond that, but then the properties of `RecursiveHierarchy` no longer work.

Is that restriction OK?